client/chat/channel.tsx

```diff
@@ -182,140 +178,121 @@ interface UserListProps {
   users: UsersRecord
 }
 
-class UserList extends React.Component<UserListProps> {
-  state = {
-    width: 0,
-    height: 0,
-  }
-  _contentRef = React.createRef<HTMLDivElement>()
-  _listRef = React.createRef<VirtualizedList>()
-
-  componentDidMount() {
-    this.updateDimensions()
-  }
+const UserList = React.memo((props: UserListProps) => {
+  const { active, idle, offline } = props.users
+  const [dimensionsRef, containerRect] = useObservedDimensions()
+  const containerRef = useRef<HTMLElement | null>(null)
```

Comment:
This is never used for anything, so I think you can remove it and just use `dimensionsRef` directly instead of doing the multiplexing function ref thing